maplibre/src/render/resource/buffer_pool.rs

```diff
@@ -446,12 +496,14 @@ pub struct IndexEntry {
     // Range of bytes within the backing buffer for indices
     buffer_indices: Range<wgpu::BufferAddress>,
     // Range of bytes within the backing buffer for metadata
-    buffer_layer_metadata: Range<wgpu::BufferAddress>,
+    buffer_layer_metadata: Option<Range<wgpu::BufferAddress>>,
     // Range of bytes within the backing buffer for feature metadata
-    buffer_feature_metadata: Range<wgpu::BufferAddress>,
+    buffer_feature_metadata: Option<Range<wgpu::BufferAddress>>,
     // Amount of actually usable indices. Each index has the size/format `IndexDataType`.
     // Can be lower than size(buffer_indices) / indices_stride because of alignment.
     usable_indices: u32,
+    // Indicate if current entry represent raster tile
+    raster: bool,
```

Comment:
I think we should not mix the raster tiles into this buffer pool.

The buffer pool is intended to be used when you have thousand of vertices. It can be used to provide views into the buffer and automatically discarding older entires

Reply:
So we need to come up with some abstraction